### PR TITLE
hotfix: #1045 use non-static get mimes types

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | yes with sf 2.8
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1045
| License       | MIT
| Doc PR        | none

### CODE REVIEW

File: src/EventListener/AddFormatListener.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function onKernelRequest(GetResponseEvent $event)
         } elseif (!isset($this->formats[$routeFormat])) {
             throw new NotFoundHttpException('Not Found');
         } else {
-            $mimeTypes = Request::getMimeTypes($routeFormat);
+            $mimeTypes = $request->getMimeTypes($routeFormat);
```
Comment (meyerbaptiste) [inline src/EventListener/AddFormatListener.php]:
```diff
- $request->getMimeTypes($routeFormat);
+ $request->getMimeType($routeFormat);
```

Comment (soyuka) [inline src/EventListener/AddFormatListener.php]: +1, though `static array getMimeTypes(string $format)` was returning an array whereas `string getMimeType(string $format)` returns a string.

File: src/EventListener/AddFormatListener.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ public function onKernelRequest(GetResponseEvent $event)
         } elseif (!isset($this->formats[$routeFormat])) {
             throw new NotFoundHttpException('Not Found');
         } else {
-            $mimeTypes = Request::getMimeTypes($routeFormat);
+            $mimeTypes = [$request->getMimeType($routeFormat)];
```
Comment (meyerbaptiste) [inline src/EventListener/AddFormatListener.php]:
```php
$mimeTypes = (array) $request->getMimeType($routeFormat);
```

Comment (soyuka) [inline src/EventListener/AddFormatListener.php]: Ouh, nice hack thanks :D. 

```
php > var_dump((array) 'teset');
array(1) {
  [0]=>
  string(5) "teset"
}
php > var_dump((array) null);
array(0) {
}
```

**Comment by meyerbaptiste** [comment]
But we can't fix this issue like that. We need all formats. For example with `xml`:
```php
(array) $request->getMimeType('xml'); 
// ['text/xml']
```
So if we have an `Accept: application/xml` header, the negotiator can't find the best format and a `NotAcceptableHttpException` exception is thrown...

```php
$negotiator->getBest('application/xml', ['text/xml']);
// NULL
```

But if we have all formats, that works because `application/xml` is in in our array:
```php
$request::getMimeTypes('xml');
// ['text/xml', 'application/xml', 'application/x-xml']
```

It's why Behat failed.

**Comment by Simperfit** [comment]
closing as won't fix

**Pull request closed without merging**